src/plugins/intel_npu/tests/functional/behavior/npuw/mocks/fake_plugins.cpp

```diff
@@ -0,0 +1,356 @@
+    const static std::vector<ov::PropertyName> roProperties{
+        RO_property(ov::supported_properties.name()),
+        RO_property(ov::available_devices.name()),
+        RO_property(ov::loaded_from_cache.name()),
+        RO_property(ov::device::uuid.name()),
+        // RO_property(METRIC_KEY(IMPORT_EXPORT_SUPPORT)),
+    };
+    // the whole config is RW before network is loaded.
+    const static std::vector<ov::PropertyName> rwProperties{
+        RW_property(ov::num_streams.name()),
+        RW_property(ov::enable_profiling.name()),
+    };
+    std::string device_id;
+    if (arguments.find(ov::device::id.name()) != arguments.end()) {
+        device_id = arguments.find(ov::device::id.name())->second.as<std::string>();
+    }
+    if (name == ov::supported_properties) {
+        std::vector<ov::PropertyName> supportedProperties;
+        supportedProperties.reserve(roProperties.size() + rwProperties.size());
+        supportedProperties.insert(supportedProperties.end(), roProperties.begin(), roProperties.end());
+        supportedProperties.insert(supportedProperties.end(), rwProperties.begin(), rwProperties.end());
+
+        return decltype(ov::supported_properties)::value_type(supportedProperties);
+    } else if (name == ov::internal::supported_properties) {
+        return decltype(ov::internal::supported_properties)::value_type(
+            {ov::PropertyName{ov::internal::caching_properties.name(), ov::PropertyMutability::RO},
+             ov::PropertyName{ov::internal::exclusive_async_requests.name(), ov::PropertyMutability::RW}});
+    } else if (name == ov::internal::exclusive_async_requests) {
+        return decltype(ov::internal::exclusive_async_requests)::value_type{exclusive_async_requests};
+    } else if (name == ov::device::uuid) {
+        ov::device::UUID uuid;
+        for (size_t i = 0; i < uuid.MAX_UUID_SIZE; i++) {
+            if (device_id == device_ids[0])
+                uuid.uuid[i] = static_cast<uint8_t>(i);
+            else if (device_id == device_ids[1])
+                uuid.uuid[i] = static_cast<uint8_t>(i * 2);
+            else if (device_id == device_ids[2])
+                uuid.uuid[i] = static_cast<uint8_t>(i * 3);
+        }
+        return decltype(ov::device::uuid)::value_type{uuid};
+    } else if (name == ov::available_devices) {
+        return decltype(ov::available_devices)::value_type(device_ids);
+    } else if (name == ov::device::capabilities) {
+        std::vector<std::string> capabilities;
+        capabilities.push_back(ov::device::capability::EXPORT_IMPORT);
+        return decltype(ov::device::capabilities)::value_type(capabilities);
+    } else if (name == "SUPPORTED_CONFIG_KEYS") {  // TODO: Remove this key
+        std::vector<std::string> configs;
+        for (const auto& property : rwProperties) {
+            configs.emplace_back(property);
+        }
+        return configs;
+//  } else if (METRIC_KEY(IMPORT_EXPORT_SUPPORT) == name) {
+//            return true;
```

Comment:
Are these you changes or it's taken as well?

Reply:
It is taken, I just refactored them a bit. However, I've preserved that comment as I don't know how much it is important.